app/screens/home/channel_list/categories_list/categories/categories.tsx

```diff
@@ -49,17 +49,25 @@ const Categories = ({
     const isTablet = useIsTablet();
     const switchingTeam = useTeamSwitch();
     const teamId = categories[0]?.teamId;
+    const [showAllCategory, setShowAllCategory] = useState(onlyUnreads);
     const categoriesToShow = useMemo(() => {
+        const orderedCategories = [...categories];
+        orderedCategories.sort((a, b) => a.sortOrder - b.sortOrder);
+
+        if (onlyUnreads && unreadsOnTop) {
+            setShowAllCategory(false);
+        } else {
+            setShowAllCategory(onlyUnreads);
+        }
```

Comment:
Why calling a `setState` inside a useMemo?

Reply:
@larkox removing it from `useMemo`, will use something like
`
  useEffect(() => {
        if (onlyUnreads && unreadsOnTop) {
            setShowAllCategory(false);
        } else {
            setShowAllCategory(onlyUnreads);
        }
    }, [onlyUnreads, unreadsOnTop, showAllCategory])
`